diff --git a/newlib/libc/machine/mips/machine/_types.h b/newlib/libc/machine/mips/machine/_types.h
new file mode 100644
index 0000000..af4caf3
--- /dev/null
+++ b/newlib/libc/machine/mips/machine/_types.h
@@ -0,0 +1,12 @@
+/*
+ *  $Id$
+ */
+
+#ifndef _MACHINE__TYPES_H
+#define _MACHINE__TYPES_H
+#include <machine/_default_types.h>
+
+typedef	__int64_t	_off_t;
+#define	__machine_off_t_defined
+
+#endif
